src/lib/support/BufBound.h

```diff
@@ -38,103 +38,128 @@ class BufBound
     }
 
-    size_t PutLE(uint64_t x, size_t size)
+    BufBound & Put(uint64_t x, size_t size)
     {
         while (size-- > 0)
         {
             uint8_t c = x & 0xff;
             Put(c);
             x >>= 8;
         }
-        return mWritten;
+        return *this;
     }
 
     /*
      * @brief write integer x into the buffer, least significant byte first
      */
-    size_t PutLE64(uint64_t x) { return PutLE(x, sizeof(x)); }
-    size_t PutLE32(uint32_t x) { return PutLE(x, sizeof(x)); }
-    size_t PutLE16(uint16_t x) { return PutLE(x, sizeof(x)); }
+    BufBound & Put64(uint64_t x) { return Put(x, sizeof(x)); }
+    BufBound & Put32(uint32_t x) { return Put(x, sizeof(x)); }
+    BufBound & Put16(uint16_t x) { return Put(x, sizeof(x)); }
 
-    size_t PutBE(uint64_t x, size_t size)
+    BufBound & PutBE(uint64_t x, size_t size)
     {
         while (size-- > 0)
         {
             uint8_t c = (x >> (size * 8)) & 0xff;
             Put(c);
         }
-        return mWritten;
+        return *this;
     }
 
     /*
      * @brief write integer x into the buffer, most significant byte first
      */
-    size_t PutBE64(uint64_t x) { return PutBE(x, sizeof(x)); }
-    size_t PutBE32(uint32_t x) { return PutBE(x, sizeof(x)); }
-    size_t PutBE16(uint16_t x) { return PutBE(x, sizeof(x)); }
+    BufBound & PutBE64(uint64_t x) { return PutBE(x, sizeof(x)); }
+    BufBound & PutBE32(uint32_t x) { return PutBE(x, sizeof(x)); }
+    BufBound & PutBE16(uint16_t x) { return PutBE(x, sizeof(x)); }
 
     /*
      * @brief append a buffer
      */
-    size_t Put(const uint8_t * buf, size_t len) { return Put(reinterpret_cast<const void *>(buf), len); }
-    size_t Put(const void * buf, size_t len)
+    BufBound & Put(const uint8_t * buf, size_t len) { return Put(reinterpret_cast<const void *>(buf), len); }
+    BufBound & Put(const void * buf, size_t len)
     {
         size_t available = Available();
 
-        memmove(mBuf + mWritten, buf, available < len ? available : len);
+        memmove(mBuf + mNeeded, buf, available < len ? available : len);
 
-        mWritten += len;
+        mNeeded += len;
 
-        return mWritten;
+        return *this;
     }
 
     /*
-     * @brief how much was or would have been written
+     * @brief number of bytes required to satisfy all calls to Put() so far
      */
-    size_t Written() { return mWritten; }
+    size_t Needed() { return mNeeded; }
     /*
      * @brief bytes available
      */
-    size_t Available() { return mLen < mWritten ? 0 : mLen - mWritten; }
+    size_t Available() { return mSize < mNeeded ? 0 : mSize - mNeeded; }
+
     /*
      * @brief whether the input fit in the buffer
      */
-    bool Fit() { return mLen >= mWritten; }
+    bool Fit()
+    {
+        size_t _;
+        return Fit(_);
+    }
+
+    /*
+     * @brief whether the input fit in the buffer, output what was
+     *        actually written
+     */
+    bool Fit(size_t & written)
```

Comment:
This is a `const` method: `bool Fit(size_t & written) const`

Also, suggest renaming argument `written` to `actually_written` perhaps.